### PR TITLE
Update Helm chart

### PR DESCRIPTION
**Pull request by matthchr**
- Use more best practices for Helm charts as defined here:
    https://helm.sh/docs/chart_best_practices/conventions/
  - Update chart to contain recently added resources.
  - Update chart version to 1.0.0, so that we can begin following semver
    more closely.

### CODE REVIEW

File: charts/azure-service-operator/requirements.lock [hunk 1]
```diff
@@ -1,6 +0,0 @@
-dependencies:
```
[the remark this answers is not in the document]
Comment (babbageclunk) [reply charts/azure-service-operator/requirements.lock:1]: Oh, I see that same information incorporated in the index.yaml now, so I think this is good.

**Review by babbageclunk — APPROVED** [review]

File: charts/azure-service-operator/requirements.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthchr) [reply charts/azure-service-operator/requirements.lock:1]: Yah actually this file name was deprecated as part of Helm 2 but we just never cleaned it up.

**Comment by matthchr** [comment]
Note that originally I updated this to use a more recent version of AAD Pod Identity but then I realized that the newer chart versions have an issue: https://github.com/Azure/aad-pod-identity/issues/885

They've already added that issue to their next milestone so when that lands we can update to the newer version.

**Comment by matthchr** [comment]
**Note**: I think it makes sense to hold off merging this PR until we're ready to actually ship a release, as otherwise it'll be broken since it won't have the latest manager instance.

**Comment by babbageclunk** [comment]
Superseded by #1332

**Pull request closed without merging**